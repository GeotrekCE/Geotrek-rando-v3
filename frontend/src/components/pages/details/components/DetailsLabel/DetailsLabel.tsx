import SVG from 'react-inlinesvg';
import parse from 'html-react-parser';
import { HtmlText } from 'components/pages/details/utils';
import { Label } from 'modules/label/interface';
import { AlertTriangle } from 'components/Icons/AlertTriangle';
import Image from 'next/image';

interface DetailsLabelProps extends Label {
  className?: string;
}

export const DetailsLabel: React.FC<DetailsLabelProps> = ({
  className = '',
  name,
  advice,
  pictogramUri,
}) => {
  if (!name) return null;

  return (
    <div
      id="details_recommandationLabel"
      className={`py-4 desktop:py-5 px-3 desktop:px-4 flex items-start text-warning
      rounded-2xl border-2 border-solid border-warning
      ${className}`}
    >
      <div className="mr-2 desktop:mr-3 shrink-0 w-6 h-6 desktop:h-12 desktop:w-12">
        {pictogramUri !== null ? <LabelIcon pictogramUri={pictogramUri} /> : <AlertTriangle />}
      </div>
      <div className="text-greyDarkColored text-Mobile-C2 desktop:text-P1">
        <div className="text-P1 desktop:text-H4 text-warning font-bold">
          <HtmlText>{parse(name)}</HtmlText>
        </div>
        {advice && <HtmlText>{parse(advice)}</HtmlText>}
      </div>
    </div>
  );
};

const LabelIcon: React.FC<{ pictogramUri: string }> = ({ pictogramUri }) => {
  if (RegExp(/(.*).svg/).test(pictogramUri)) {
    return <SVG src={pictogramUri} className="w-6 h-6 desktop:w-10 desktop:h-10" />;
  }
  return (
    <Image
      loading="lazy"
      className="object-center object-cover w-6 h-6 desktop:h-10 desktop:w-10 rounded-full"
      src={pictogramUri}
      width={24}
      height={24}
      alt=""
    />
  );
};
