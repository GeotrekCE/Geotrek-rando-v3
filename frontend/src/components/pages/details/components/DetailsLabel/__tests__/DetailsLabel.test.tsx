import { render } from 'services/testing/reactTestingLibraryWrapper';
import { DetailsLabel } from '../DetailsLabel';
describe('DetailsLabel', () => {
  it('should display a DetailsLabel', () => {
    const propsLabel = {
      id: 1,
      advice:
        'Le Parc national est un territoire naturel, ouvert à tous, mais soumis à une réglementation qu’il est utile de connaître pour préparer son séjour',
      name: 'En coeur de parc',
      pictogramUri: 'https://geotrekdemo.ecrins-parcnational.fr/media/upload/desktype-info.svg',
      className: '',
    };
    const component = render(<DetailsLabel {...propsLabel} />);

    expect(component).toMatchSnapshot();
  });
});
