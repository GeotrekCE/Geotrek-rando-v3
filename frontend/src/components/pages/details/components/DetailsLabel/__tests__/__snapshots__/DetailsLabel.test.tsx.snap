// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsLabel should display a DetailsLabel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-4 desktop:py-5 px-3 desktop:px-4 flex items-start text-warning
      rounded-2xl border-2 border-solid border-warning
      "
        id="details_recommandationLabel"
      >
        <div
          class="mr-2 desktop:mr-3 flex-shrink-0 w-6 h-6 desktop:h-12 desktop:w-12"
        />
        <div
          class="text-greyDarkColored text-Mobile-C2 desktop:text-P1"
        >
          <div
            class="text-P1 desktop:text-H4 text-warning font-bold"
          >
            <div
              class="sc-bdfBQB hRTanl"
            >
              En coeur de parc
            </div>
          </div>
          <div
            class="sc-bdfBQB hRTanl"
          >
            Le Parc national est un territoire naturel, ouvert à tous, mais soumis à une réglementation qu’il est utile de connaître pour préparer son séjour
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-4 desktop:py-5 px-3 desktop:px-4 flex items-start text-warning
      rounded-2xl border-2 border-solid border-warning
      "
      id="details_recommandationLabel"
    >
      <div
        class="mr-2 desktop:mr-3 flex-shrink-0 w-6 h-6 desktop:h-12 desktop:w-12"
      />
      <div
        class="text-greyDarkColored text-Mobile-C2 desktop:text-P1"
      >
        <div
          class="text-P1 desktop:text-H4 text-warning font-bold"
        >
          <div
            class="sc-bdfBQB hRTanl"
          >
            En coeur de parc
          </div>
        </div>
        <div
          class="sc-bdfBQB hRTanl"
        >
          Le Parc national est un territoire naturel, ouvert à tous, mais soumis à une réglementation qu’il est utile de connaître pour préparer son séjour
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
